datalake/src/main/java/com/sequenceiq/datalake/cm/RangerCloudIdentityService.java

```diff
@@ -64,11 +70,20 @@ private RangerCloudIdentitySyncStatus setAzureCloudIdentityMapping(String envCrn
         return Optional.of(Iterables.getOnlyElement(sdxClusters));
     }
 
+    private boolean isDatalakeRunning(SdxCluster sdxCluster) {
+        SdxStatusEntity sdxStatusEntity = sdxStatusService.getActualStatusForSdx(sdxCluster);
+        LOGGER.debug("SDX status = {}", sdxStatusEntity.getStatus());
+        return sdxStatusEntity.getStatus().equals(DatalakeStatusEnum.RUNNING);
+    }
+
     public RangerCloudIdentitySyncStatus setAzureCloudIdentityMapping(String envCrn, Map<String, String> azureUserMapping) {
         Optional<SdxCluster> sdxCluster = getSdxCluster(envCrn);
         if (sdxCluster.isEmpty()) {
             return newSyncStatus(RangerCloudIdentitySyncState.NOT_APPLICABLE, "No datalakes associated with the environment.");
         }
+        if (!isDatalakeRunning(sdxCluster.get())) {
+            return newSyncStatus(RangerCloudIdentitySyncState.NOT_APPLICABLE, "Datalake is not running for the environment.");
+        }
```

Comment:
this is a duplicate of the `getRangerCloudIdentitySyncStatus`. could you refactor this to a common part to reduce code duplication?

Reply:
Thanks, I've added the `else if` branches as per your chat suggestion. I think making it more DRY would require wrapping the operation in a functional interface (see below example) which seemed less readable to me, so I didn't go for it. Let me know if that's okay, or if you had some other idea.


```java
public RangerCloudIdentitySyncStatus runCloudIdSyncOperation(String envCrn, Function<SdxCluster, RangerCloudIdentitySyncStatus> operation) {
    Optional<SdxCluster> sdxCluster = getSdxCluster(envCrn);
    if (sdxCluster.isEmpty()) {
        return newSyncStatus(RangerCloudIdentitySyncState.NOT_APPLICABLE, "No datalakes associated with the environment.");
    } else if (!isDatalakeRunning(sdxCluster.get())) {
        return newSyncStatus(RangerCloudIdentitySyncState.NOT_APPLICABLE, "Datalake is not running for the environment.");
    } else {
        return operation.apply(sdxCluster.get());
    }
}

public RangerCloudIdentitySyncStatus setAzureCloudIdentityMapping(String envCrn, Map<String, String> azureUserMapping) {
    return runCloudIdSyncOperation(envCrn, sdxCluster -> setAzureCloudIdentityMapping(envCrn, sdxCluster, azureUserMapping));
}

public RangerCloudIdentitySyncStatus getRangerCloudIdentitySyncStatus(String envCrn, long commandId) {
    return runCloudIdSyncOperation(envCrn, sdxCluster -> {
        try {
            ApiCommand apiCommand = clouderaManagerRangerUtil.getApiCommand(sdxCluster.getStackCrn(), commandId);
            return toRangerCloudIdentitySyncStatus(apiCommand);
        } catch (ApiException e) {
            LOGGER.error("Encountered cloudera manager api exception", e);
            return newSyncStatus(RangerCloudIdentitySyncState.FAILED, "Encountered cloudera manager api exception");
        }
    });
}
```